spec/models/commodity_spec.rb

```diff
@@ -825,4 +825,51 @@
       it { is_expected.not_to be_fast_declarable }
     end
   end
+
+  describe '#special_nature?' do
+    context 'when commodity has measures with atleast one measure condition containing special nature certificate' do
+      let(:commodity) { create(:commodity) }
+
+      before do
+        create(
+          :measure,
+          :with_base_regulation,
+          :with_measure_conditions,
+          :with_special_nature,
+          goods_nomenclature_sid: commodity.goods_nomenclature_sid,
+        )
+      end
+
+      it { expect(commodity.special_nature?).to eq(true) }
+    end
+
+    context 'when commodity does not have measures with atleast one measure condition containing special nature certificate' do
```

Comment:
do you mean 
`when commodity has no measure conditions containing special nature certificate`
?